pwiz/data/msdata/mzmlb/Connection_mzMLb.cpp

```diff
@@ -0,0 +1,612 @@
+                    std::string version = ver;
+                    if (version != "mzMLb 0.6")
+                    {
+                        H5Aclose(aid);
+                        H5Aclose(atype_mem);
+                        close();
+                        throw std::runtime_error("[Connection_mzMLb::open()] We cannot read this version of mzMLb file.");
+                    }
+                   
+                }
+                H5Aclose(atype_mem);
+            }
+            H5Aclose(atype);
+        }
+        H5Aclose(aid);        
+    }
+    H5Pclose(dapl);
+
+    if (identifyOnly)
+    {
+        close();
+        return;
+    }
+
+    opaque_id_ = H5Tcreate(H5T_OPAQUE, 1);
+ }
+
+
+Connection_mzMLb::Connection_mzMLb(const std::string& id, int chunk_size, int compression_level) :
+    chunk_size_(chunk_size),
+    compression_level_(compression_level)
+{    
+    H5Eset_auto(H5E_DEFAULT, NULL, NULL);
+
+    // create/truncate HDF5 file for writing
+    hid_t fapl = H5Pcreate(H5P_FILE_ACCESS);
+    {
+        int nelemts; // Dummy parameter in API, no longer used
+        size_t nslots; // Number of slots in the hash table
+        size_t nbytes; // Size of chunk cache in bytes
+        double w0; // Chunk preemption policy
+        H5Pget_cache(fapl, &nelemts, &nslots, &nbytes, &w0);
+        nbytes = (nbytes > chunk_size) ? nbytes : chunk_size; // Set per dataset cache to twice the chunk size please
+        w0 = 1.0; // since pwiz only writes a spectrum once
+        H5Pset_cache(fapl, nelemts, nslots, nbytes, w0);
+        file_ = H5Fcreate(id.c_str(), H5F_ACC_TRUNC, H5P_DEFAULT, fapl);
+        if (file_ < 0)
+        {
+            H5Pclose(fapl);
+            throw std::runtime_error("[Connection_mzMLb::Connection_mzMLb()] Could not open or create mzMLb file for writing.");
+        }
+
+        // create dataset to store mzML XML
+        hid_t dcpl = H5Pcreate(H5P_DATASET_CREATE);
+        {        
+            hsize_t cdims = chunk_size;
+            H5Pset_chunk(dcpl, 1, &cdims);
+            if (compression_level > 0)
+            {
+                hsize_t level = compression_level;
+                H5Pset_deflate(dcpl, level);
+            }
+            H5Pset_fletcher32(dcpl);
+            hsize_t maxdims = H5S_UNLIMITED;
+            mzML_.space = H5Screate_simple(1, &mzML_.size, &maxdims);
+            mzML_.dataset = H5Dcreate(file_, "mzML", H5T_NATIVE_CHAR, mzML_.space, H5P_DEFAULT, dcpl, H5P_DEFAULT);
+        }
+        H5Pclose(dcpl);
+
+        // write version string
+        hid_t aid = H5Screate(H5S_SCALAR);
+        {
+            hid_t atype = H5Tcopy(H5T_C_S1);
+            {
+                H5Tset_size(atype, 10);
+                H5Tset_strpad(atype, H5T_STR_NULLTERM);
+                hid_t attrid = H5Acreate(mzML_.dataset, "version", atype, aid, H5P_DEFAULT, H5P_DEFAULT);
+                {
+                    std::string version_out = "mzMLb 0.6";
```

Comment:
Again, is this the value going forward? This should probably be factored out as class static member or macro-defined?